src/main/java/datawave/query/util/DateFrequencyValue.java

```diff
@@ -0,0 +1,359 @@
+package datawave.query.util;
+
+import org.apache.accumulo.core.data.Value;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * This class handles the serialization and deserialization of the Accumulo value in a record of the Datawave Metadata table that has a column family of "f" and
+ * a column qualifier that is prefixed with the string "compressed-" like "compressed-csv" for example. This is a class used to help compress the date and
+ * frequency values that are aggregated together to by the FrequencyTransformIterator and manipulated in the FrequencyFamilyCounter The byte array really only
+ * has to be like this in regular expression format (YEAR(4BYTE-FREQUENCY){365,366})(\x00))* . Explained verbally a one byte representation of Year followed by
+ * 365 or 366 (Leap year) 4 byte holders for frequency then Null terminated. Each Accumulo row for this "aggregated" frequency "map" would be 10 x ( 1 + (365 x
+ * 4) + 1) bytes long for a 10 year capture: 14620 bytes.
+ */
+
+public class DateFrequencyValue {
+    
+    private static final Logger log = LoggerFactory.getLogger(DateFrequencyValue.class);
+    
+    // An intermediate data structure build up by iterating through uncompressDateFrequencies
+    private HashMap<CompressedMapKey,byte[]> compressedDateFrequencies = new HashMap<>();
+    // The mapping of SimpleDateFormat strings to frequency counts. This is passed in by the
+    // FrequencyFamilyCounter object to this classes serialize function
+    private HashMap<String,Integer> uncompressedDateFrequencies = null;
+    private static int BADORDINAL = 367;
+    private static int DAYS_IN_LEAP_YEAR = 6;
+    private static int MAX_YEARS = 10;
+    private static int NUM_BYTES_YR = 4;
+    
+    public DateFrequencyValue() {}
+    
+    /**
+     *
+     * @param dateToFrequencyValueMap
+     *            the keys should be dates in yyyyMMdd format
+     * @return Value the value to store in accumulo
+     */
+    public Value serialize(HashMap<String,Integer> dateToFrequencyValueMap, boolean compress) {
+        
+        Value serializedMap;
+        uncompressedDateFrequencies = dateToFrequencyValueMap;
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        
+        if (compress) {
```

Comment:
I don't get the compress flag here.  If compress=false, then this method will return a Value with a length of zero.

Reply:
perhaps compress should denote whether we gzip or not?

Reply:
I will change the name of that boolean  from "compress" to "doGzip" .    I will make that change now and push it.